index.js

```diff
@@ -15,16 +15,16 @@ module.exports = fp(function (fastify, options, next) {
 
     // static salt to be used for key derivation, not great for security,
     // but better than nothing
-    var salt = 'mq9hDxBVDbspDR6nLfFT1g=='
+    var salt = 'mq9hDxBVDbspDR6n'
 
     if (options.salt) {
       salt = options.salt
     }
 
-    salt = Buffer.from(salt, 'base64')
+    salt = Buffer.from(salt, 'ascii')
 
     if (Buffer.byteLength(salt) !== sodium.crypto_pwhash_SALTBYTES) {
-      return next(new Error('salt must be length 24'))
+      return next(new Error('salt must be length 16'))
```

Comment:
I would use the value from `sodium.crypto_pwhash_SALTBYTES` to avoid future issues.

Reply:
I've used `sodium.crypto_pwhash_SALTBYTES` but after refactoring I forget it. I will fix it.